src/cppapi/position.cpp

```diff
@@ -0,0 +1,60 @@
+
+#include "openmc/position.h"
+#include "openmc/cppapi/position.h"
+#include "openmc/capi.h"
+#include "openmc/error.h"
+
+#include <pybind11/operators.h>
+
+#include <stdexcept>
+#include <vector>
+
+
+namespace openmc {
+
+using std::vector;
+namespace py = pybind11;
+
+//==============================================================================
+// CPP API
+//==============================================================================
+//Add elements of Position to the module
+void init_position(py::module& m)
+{
+  py::class_<Position> (m,"Position")
+    .def(py::init<double, double, double>())
+    .def(py::init<const vector<double>& >())
+    .def(py::init<const std::array<double, 3>&>())
```

Comment:
Let's just keep the first constructor in the bindings (double, double, double). If a Python user wants to pass an iterable, they can unpack it with `*`.